exporter/sapmexporter/factory.go

```diff
@@ -15,23 +15,24 @@
 package sapmexporter
 
 import (
+	"context"
+
 	"github.com/open-telemetry/opentelemetry-collector/component"
 	"github.com/open-telemetry/opentelemetry-collector/config/configerror"
 	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
-	"go.uber.org/zap"
 )
 
 const (
 	// The value of "type" key in configuration.
-	typeStr = "sapm"
+	typeStr = configmodels.Type("sapm")
```

Comment:
This is not necessary and it will require less changes elsewhere if you keep `typeStr = "sapm"`.